app/src/main/kotlin/com/exallium/todoapp/editnote/EditNotePresenter.kt

```diff
@@ -30,6 +30,24 @@ class EditNotePresenter(private val view: EditNoteView,
         setupSaveNoteSubscription(noteId)
 
         view.cancelEditNoteClicks().map { null }.subscribe(showNewNoteDetailSubscriberFn).addToComposite()
+
+        setupTextViewsChanged()
+    }
+
+    fun setupTextViewsChanged() {
+        view.titleTextChanges()
+                .map { model.validateNoteTitleText(view.getNewNoteTitle()) }
+                .doOnNext { if (!it) { view.showInvalidNoteTitleError(); view.toggleSubmit(false) } }
+                // filters out anything that doesn't match the predicate (only emit true events (the text is valid).
+                .filter { it }
+                .subscribe { view.toggleSubmit(validateAllFields()) }
+
+        view.bodyTextChanges()
```

Comment:
could you make a follow-up commit with how you'd improve it so I can learn from it?